pkg/encoding/brotli/brotli.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package brotli
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/tricksterproxy/trickster/pkg/encoding/reader"
+
+	"github.com/andybalholm/brotli"
+)
+
+// Decode returns the inflated version of the gzip-deflated byte slice
+func Decode(in []byte) ([]byte, error) {
+	br := brotli.NewReader(bytes.NewReader(in))
+	return io.ReadAll(br)
+}
+
+// Encode returns the gzip-deflated version of the byte slice
+func Encode(in []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
```

Comment:
Maybe it's silly but you could save some rummaging around in the buffer code if you change this to
`buf := bytes.NewBuffer(len(in))`. It's not a big deal but you get to skip a couple of lines since you already know the size of the thing you are writing.